src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -921,6 +928,44 @@ impl FlowService for FlowServiceImpl {
         })
     }
 
+    /// Returns states of flows associated with a given account
+    /// ordered by creation time from newest to oldest
+    /// Applies specified filters
+    #[tracing::instrument(level = "debug", skip_all, fields(%account_id, ?filters, ?pagination))]
+    async fn list_all_flows_by_account(
+        &self,
+        account_id: &AccountName,
+        filters: DatasetFlowFilters,
+        pagination: FlowPaginationOpts,
+    ) -> Result<FlowStateListing, ListFlowsByDatasetError> {
+        let total_count = self
+            .flow_event_store
+            .get_count_flows_by_account(account_id, &filters)
+            .await
+            .int_err()?;
+        let account_id = account_id.clone();
+
+        let matched_stream = Box::pin(async_stream::try_stream! {
+            let relevant_flow_ids: Vec<_> = self
+                .flow_event_store
+                .get_all_flow_ids_by_account(&account_id, filters, pagination)
+                .try_collect()
+                .await
+                .int_err()?;
+
+            // TODO: implement batch loading
+            for flow_id in relevant_flow_ids {
+                let flow = Flow::load(flow_id, self.flow_event_store.as_ref()).await.int_err()?;
+                yield flow.into();
+            }
```

Comment:
Yeah, batch loading is kind of important. Not a part of this ticket, but would be important very soon.